### ISSUE TITLE
Save Bug

### ISSUE THREAD

**Issue by sj365** [gildas-lormeau/SingleFile#232]

**Comment by sj365** [comment]
Environment win 10 Pro x64 up to date
Chrome, Opera, browsers up to date
Firefox 52.8.1
SingleFileZ deault values


SingleFileZ current version does not accurately save the following pages

https://guides.github.com/activities/hello-world/

right side text formatting is not anchored correctly

https://www.thedailybeast.com/when-they-see-us-central-park-five-prosecutor-takes-fire-over-ava-duvernay-netflix-series?ref=wrap

I am using ublock origin but the site displays correctly in all browsers.  When i try to save with SingleFileZ and SingleFile, the method used seems to interact with the page that causes it to be totally blank and the saved html file is also blank in all browsers once opened.  As a test I tried to save the same page with Save Page WE and it was able to do it 100% correctly once opened with all browsers.

SingleFileZ and SingleFile also has problems saving long pages in the washingtonpost.com - if comment section is expanded to 10-20 comments the program might not save file on first try.  Also it will save the page and the comments section will be left out entirely.  This is both with and without the use of helper news site bypass scripts active.  

The scripts are not the issue as all sites, pages, and browsers have been tested, saved, and display properly with Mozilla Archive Format which I still use in Firefox because it is the only application that has consistently worked in all sites 100%.  I have spent/wasted this weekend testing almost all the related extensions in Chrome Store and Firefox to try and find a current replacement for MAF; over 20 tested and only Save Page WE and Single File come close to consistently working.  The main webpages i have been using to test these applications in all browsers listed are:

https://www.thedailybeast.com/when-they-see-us-central-park-five-prosecutor-takes-fire-over-ava-duvernay-netflix-series?ref=wrap

https://pagesix.com/2019/06/21/venus-williams-and-boyfriend-nicky-hammond-split-up/

https://www.nytimes.com/2018/08/10/science/eugene-parker-solar-wind-nasa-probe.html

https://guides.github.com/activities/hello-world/

SingleFile and SingleFileZ can not save all the above sites %100 correct (Neither can Save Page WE).  Picture elements, headers, or alignment issues degrade the saved file.  Again, MAF can save all of them perfectly as mhtml.  Maybe some code tricks can be salvaged from it to help SingleFile.

**Comment by gildas-lormeau** [comment]
Please disable the option "save deferred images" and let me know which saved pages are incorrect. On my end, all these URLs are saved correctly.

**Comment by gildas-lormeau** [comment]
I'm sorry, I just understood what you meant by "right side text formatting is not anchored correctly". Actually, you're right and this particular issue is not related to the save deferred images" option. It may be related to the fact that the JavaScript is removed from the saved page though.

**Comment by gildas-lormeau** [comment]
@sj365 I did some tests and I confirm that disabling the option "Other resources > remove scripts" fixes the issues related to anchored menus and headers on https://guides.github.com/activities/hello-world, https://www.thedailybeast.com/when-they-see-us-central-park-five-prosecutor-takes-fire-over-ava-duvernay-netflix-series?ref=wrap and https://pagesix.com/2019/06/21/venus-williams-and-boyfriend-nicky-hammond-split-up. This option is enabled by default because I cannot guarantee it will work and because it can lead to requests sent to third-party domains when displaying the saved page.

**Comment by sj365** [comment]
Thanks for replying.  I didnt know where my post was when i came back to see if there was a response.  Im new to this site and not familiar with where things are.  

Helper info - if you're interested, 2 things to help evaluate pages that use them; Anti-Paywall and Burlesco both on Github.com  Used simultaneously, many sites can be viewed for testing purposes.

https://abc30.com/new-details-released-on-central-fresno-shooting-that-critically-injured-10-month-old-girl/5360008/
https://abc7ny.com/13-year-old-charged-in-attack-on-mom-teen-in-new-jersey/5363682/
1st video space is outsized obscuring text

As far as disabling Other Resources-are you saying this is the only way to get the page correctly?  I dont want to do it if there are negative security implications.  Do you have another option?

**Comment by sj365** [comment]
I made a video to show what im still seeing with singlefile and singlefilez
https://drive.google.com/open?id=1bgXQ9RIERU-yxEuuDWN423gnYYZu_oOY

singlefile and singlefilez can save the page but wont if the banner is shown

Opera Version:60.0.3255.170
Google Chrome is up to date  Version 75.0.3770.100 (Official Build) (64-bit)

**Comment by gildas-lormeau** [comment]
@sj365 This particular issue on https://www.thedailybeast.com/ related to save "deffered images", I'll try to see what's wrong. I don't want to implement a "auto-scroll" feature...

**Comment by sj365** [comment]
thanks for checking this out.  Also, I recommend to use APW and Burlesco together.  They complement each other.

**Comment by gildas-lormeau** [comment]
@sj365 I have found the cause of the issue on https://www.thedailybeast.com. It's because when SingleFile triggers the loading of deferred contents (e.g. images), the website tries to read the cookies. However, SingleFile blocks the access to cookies during the loading of deferred images because most of the time if the website can read/write cookies this leads to a "register" popup or something similar appearing in the saved page.
To circumvent this issue, I added a hidden option allowing you to disable this feature. To toggle it, you will have to export your settings, open the exported settings in a text editor, replace `loadDeferredImagesBlockCookies: true` with `loadDeferredImagesBlockCookies: false,`, save the file and import the updated file. This will be available in the next version.
